core/dbt/parser/manifest.py

```diff
@@ -599,6 +652,173 @@ def write_manifest_for_partial_parse(self):
+
+        # build publication metadata
+        metadata = PublicationMetadata(
+            adapter_type=self.root_project.credentials.type,
+            quoting=self.root_project.quoting,
+        )
+
+        # get a list of public model ids first so it can be used in constructing dependencies
+        public_model_ids = []
+        for node in self.manifest.nodes.values():
+            if node.resource_type == NodeType.Model and node.access == AccessType.Public:
+                public_model_ids.append(node.unique_id)
+
+        set_of_public_unique_ids = set(public_model_ids)
+
+        # Get the Graph object from the Linker
+        from dbt.compilation import Linker
+
+        linker = Linker()
+        graph = linker.get_graph(self.manifest)
+
+        public_models = {}
+        for unique_id in public_model_ids:
+            model = self.manifest.nodes[unique_id]
+            # public_node_dependencies is the intersection of all parent nodes plus public nodes
+            public_node_dependencies = []
+            parents: set = graph.select_parents({unique_id})
+            public_node_dependencies = parents.intersection(set_of_public_unique_ids)
+
+            public_model = PublicModel(
+                name=model.name,
+                package_name=model.package_name,
+                unique_id=model.unique_id,
+                relation_name=model.relation_name,
+                version=model.version,
+                latest_version=model.latest_version,
+                public_node_dependencies=list(public_node_dependencies),
+                generated_at=metadata.generated_at,
+            )
+            public_models[unique_id] = public_model
+
+        dependencies = []
+        if self.manifest.project_dependencies:
+            for project in self.manifest.project_dependencies.projects:
+                dependencies.append(project.name)
+        publication = PublicationArtifact(
+            metadata=metadata,
+            project_name=self.root_project.project_name,
+            public_models=public_models,
+            dependencies=dependencies,
+        )
+        # write out publication artifact <project_name>_publication.json
+        publication_file_name = f"{self.root_project.project_name}_publication.json"
+        path = os.path.join(self.root_project.target_path, publication_file_name)
+        publication.write(path)
+
+    def build_public_nodes(self) -> bool:
+        """This method loads the dependencies from dependencies.yml, reads in the
+        the publication artifacts and adds the PublicModels to the manifest
+        "public_nodes" dictionary. It returns a boolean that indicates that
+        public nodes have been rebuilt."""
+        public_nodes_changed = False
+
+        # Load the dependencies from the dependencies.yml file
+        dependencies_filepath = resolve_path_from_base(
+            "dependencies.yml", self.root_project.project_root
```

Comment:
nit: should use DEPENDENCIES_FILE_NAME

Reply:
Moved some file_name constants to core/dbt/constants.py and used DEPENDENCIES_FILE_NAME